spring-kafka-docs/src/main/antora/modules/ROOT/pages/kafka/annotation-error-handling.adoc

```diff
@@ -242,6 +242,13 @@ public DefaultErrorHandler errorHandler(ConsumerRecordRecoverer recoverer) {
 }
 ----
 
+[IMPORTANT]
+The `DefaultErrorHandler` only processes exceptions that inherit from `RuntimeException`.
+Exceptions inheriting from `Error` bypass the error handler entirely, causing the consumer to terminate immediately, close the Kafka connection, and skip all retry/recovery mechanisms.
+This critical distinction means applications may report healthy status despite having terminated consumers that no longer process messages.
+Always ensure that exceptions thrown in message processing code extend from `RuntimeException` rather than `Error` to allow proper error handling.
```

Comment:
I’m not sure how to help, but this is not true. The `OutOfMemoryError` is really out of our control . Do you have any advice how really we can make `Error` as `RuntimeException`? What is the goal of this sentence if we don’t provide any example how to do that ?

Reply:
I meant any exceptions thrown explicitly by the applications. I can rephrase if needed. Things like OOME are outside the application's control, and we can't help. This issue was raised because someone was accidentally placed Kotlin's TODO method that throws an `Error`. This clarification is basically to navigate the users of those scenarios.